src/GraphBLAS-sharp/MatrixAndVector.fs

```diff
@@ -37,32 +39,56 @@ and [<AbstractClass>] Vector<'a when 'a : struct and 'a : equality>() =
     abstract Length: int
     abstract AsArray: 'a[]
 
-    abstract Item: Mask1D<'a> -> Vector<'a> with get, set
-    abstract Item: int * int -> Scalar<'a> with get, set
-    abstract Item: Mask1D<'a> -> Scalar<'a> with set
+    abstract CreateMask: bool -> Mask1D
 
-    abstract Vxm: Matrix<'a> -> Mask1D<'a> -> Semiring<'a> -> Vector<'a>
-    abstract EWiseAdd: Vector<'a> -> Mask1D<'a> -> Semiring<'a> -> Vector<'a>
-    abstract EWiseMult: Vector<'a> -> Mask1D<'a> -> Semiring<'a> -> Vector<'a>
-    abstract Apply: Mask1D<'a> -> UnaryOp<'a, 'b> -> Vector<'b>
+    abstract Item: Mask1D -> Vector<'a> with get, set
+    abstract Item: int -> Scalar<'a> with get, set
+    abstract Item: Mask1D -> Scalar<'a> with set
+
+    abstract Vxm: Matrix<'a> -> Mask1D -> Semiring<'a> -> Vector<'a>
+    abstract EWiseAdd: Vector<'a> -> Mask1D -> Semiring<'a> -> Vector<'a>
+    abstract EWiseMult: Vector<'a> -> Mask1D -> Semiring<'a> -> Vector<'a>
+    abstract Apply: Mask1D -> UnaryOp<'a, 'b> -> Vector<'b>
     abstract Reduce: Monoid<'a> -> Scalar<'a>
 
-    abstract EWiseAddInplace: Vector<'a> -> Mask1D<'a> -> Semiring<'a> -> unit
-    abstract EWiseMultInplace: Vector<'a> -> Mask1D<'a> -> Semiring<'a> -> unit
-    abstract ApplyInplace: Mask1D<'a> -> UnaryOp<'a, 'b> -> unit
+    abstract EWiseAddInplace: Vector<'a> -> Mask1D -> Semiring<'a> -> unit
+    abstract EWiseMultInplace: Vector<'a> -> Mask1D -> Semiring<'a> -> unit
+    abstract ApplyInplace: Mask1D -> UnaryOp<'a, 'b> -> unit
 
     static member inline (+) (x: Vector<'a>, y: Vector<'a>) = x.EWiseAdd y
     static member inline (*) (x: Vector<'a>, y: Vector<'a>) = x.EWiseMult y
     static member inline (+.*) (x: Vector<'a>, y: Matrix<'a>) = x.Vxm y
     static member inline (.+) (x: Vector<'a>, y: Vector<'a>) = x.EWiseAddInplace y
     static member inline (.*) (x: Vector<'a>, y: Vector<'a>) = x.EWiseMultInplace y
 
-and Mask1D<'a when 'a : struct and 'a : equality> =
-    | Mask1D of Vector<'a>
-    | Complemented1D of Vector<'a>
-    | None
+and Mask1D(isNone: bool, indices: int[], length: int, isRegular: bool) =
```

Comment:
Regular mask is its default state. So, I think, boolean flag should be named `isComplemented`.